python/src/robyn/modeling/pareto/immediate_carryover_calculator.py

```diff
@@ -0,0 +1,108 @@
+# pyre-strict
+
+from typing import Optional, Dict, Any
+import pandas as pd
+from robyn.data.entities.mmmdata import MMMData
+from robyn.modeling.entities.modeloutputs import ModelOutputs
+
+
+class ImmediateCarryoverCalculator:
+    """
+    A class to calculate immediate and carryover effects for media channels.
+
+    This class handles the calculation of immediate and carryover effects for different
+    media channels, including data preparation, decomposition, and result formatting.
+
+    Attributes:
+        input_collect (MMMData): Input data for the marketing mix model.
+        output_collect (ModelOutputs): Output data from the model runs.
+    """
+
+    def __init__(self, input_collect: MMMData, output_collect: ModelOutputs):
```

Comment:
please use meaningful and relevant variable names. For example, mmmdata and model_outputs.